lib/matplotlib/tests/test_image.py

```diff
@@ -853,6 +853,14 @@ def test_mask_image():
     ax2.imshow(A, interpolation='nearest')
 
 
+def test_mask_image_all():
+    # Test behavior with an image that is entirely masked does not warn
+    data = np.full((2, 2), np.nan)
+    fig, ax = plt.subplots()
+    ax.imshow(data)
+    fig.canvas.draw_idle()  # would emit a warning
```

Comment:
Unless we run with warnings as error (which I'm pretty sure we don't) this test doesn't actually check that we don't issue warnings...

Reply:
I think it failed on master before the other changes in this PR, and this suggested to me that you do:

https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/testing/conftest.py#L20

But I can wrap this in a `with pytest.warns(None) as w: ... assert len(w) == 0` if it'll help

Reply:
(see the traceback in the description, that was from the original test on `master` without the fixing-changes)

Reply:
Pytest does treat warnings as errors.

Reply:
Ah, I wasn't aware we had that turned on in `conftest.py`. I'm used to testing on MetPy where so many upstream dependencies have warnings at times that running with warnings is impractical. Apologies for the noise.